### PR TITLE
Add java code sample for tests

### PR DESCRIPTION
**Pull request by amoghrajesh**
- Added a java file with some code logic
- Updated the tests in analyzer and scanner

### CODE REVIEW

**Comment by amoghrajesh** [comment]
@JoaoDanielRufino can you help with a review? I think a UT is failing btw. Not sure why though

**Comment by amoghrajesh** [comment]
@JoaoDanielRufino i have refactored the code a bit because the tests were doing an index wise comparison which is not the best style and is not extendable too. I have done some refactoring around that end

File: test/fixtures/code_samples/_main.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package test.fixtures.code_samples;
```
Comment (JoaoDanielRufino) [inline test/fixtures/code_samples/_main.java:1]: Also, could you add a block comment on this file, for tests purpose ?

**Review by JoaoDanielRufino — CHANGES_REQUESTED** [review]

File: pkg/scanner/scanner_test.go [hunk 2]
```diff
@@ -88,3 +96,23 @@ func getExtensionsMap(languages language.Languages) map[string]string {
...
+		for _, file := range files {
+			if file == want {
+				present = true
+			}
+		}
+		globalPresent = (present == true)
+	}
+
+	return globalPresent
+}
```
[the remark this answers is not in the document]
Comment (amoghrajesh) [reply pkg/scanner/scanner_test.go:118]: Oh, good catch. Made the change

File: test/fixtures/code_samples/_main.java (hunk 1 above)
Comment (amoghrajesh) [reply test/fixtures/code_samples/_main.java:1]: @JoaoDanielRufino I am not sure if i entirely got you. You want me to make line 1 as below?
```
/*package test.fixtures.code_samples;*/
```

Comment (amoghrajesh) [reply test/fixtures/code_samples/_main.java:1]: Okay it made sense to me now. I have pushed it

File: pkg/scanner/scanner_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoaoDanielRufino) [reply pkg/scanner/scanner_test.go:118]: Please, update the code here with the suggested change

**Review by JoaoDanielRufino — CHANGES_REQUESTED** [review]

File: pkg/scanner/scanner_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (amoghrajesh) [reply pkg/scanner/scanner_test.go]: Made the changes

File: pkg/scanner/scanner_test.go [hunk 3]
```diff
@@ -44,20 +44,27 @@ func TestScan(t *testing.T) {
 		},
 		{
 			Metadata:   files[2],
+			Lines:      20,
+			CodeLines:  15,
+			BlankLines: 4,
+			Comments:   4,
```
Comment (JoaoDanielRufino) [inline pkg/scanner/scanner_test.go:50]: The expected values are incorrect, please run the test and verify if the suggested value is correct

```suggestion
			Lines:      24,
			CodeLines:  15,
			BlankLines: 5,
			Comments:   4,
```

**Review by JoaoDanielRufino — CHANGES_REQUESTED** [review]

File: pkg/scanner/scanner_test.go (hunk 3 above)
Comment (amoghrajesh) [reply pkg/scanner/scanner_test.go]: Ah good catch. Had missed it due to rebasing

**Review by JoaoDanielRufino — APPROVED** [review]

**Pull request merged**